### ISSUE TITLE
Subclass RuntimeException in StandardSchedulerClient for better error handling

### ISSUE THREAD

**Issue by virtual-machinist** [kagkarlsson/db-scheduler#167]
Due to `StandardSchedulerClient` throwing plain `RuntimeException` instances there is no way to distinguish (and possibly ignore) exceptions other than by doing string matching of the supplied message:
```
        public void cancel(TaskInstanceId taskInstanceId) {
...
            if (execution.isPresent()) {
                if (execution.get().isPicked()) {
                    throw new RuntimeException(String.format("Could not cancel schedule, the execution with name '%s' and id '%s' is currently executing", taskName, instanceId));
                }
...
            } else {
                throw new RuntimeException(String.format("Could not cancel schedule - no task with name '%s' and id '%s' was found.", taskName, instanceId));
            }
        }
```
Almost the same is done in `reschedule()`, except for slightly different message wording.

Did you consider having `TaskInstanceAlreadyRunningException` and `TaskInstanceNotFoundException` runtime exception classes? The messages can stay the same in case someone is already doing string matching for that purpose.

**Comment by virtual-machinist** [comment]
Right now I'm able to work around this issue by upgrading to the latest version (was running 9.1 where there was no `ScheduledExecution#isPicked()` method, so I had no other way that wasn't ugly and/or tedious), but a slightly richer exception model would be nice.

**Comment by kagkarlsson** [comment]
Good thing v9.2 was helpful.

I agree that replacing the generic `RunTimeException` would be helpful here, I will put it on the backlog 👍

**Comment by zendern** [comment]
@virtual-machinist @kagkarlsson take a peek at #199 i took a stab at implementing these custom exceptions.

**Comment by kagkarlsson** [comment]
🎉 This issue has been resolved in `v16.9.0` ([Release Notes](https://github.com/kagkarlsson/db-scheduler/releases/tag/v16.9.0))
